electrumsv/gui/qt/network_dialog.py

```diff
@@ -1290,7 +1292,7 @@ def _on_menu_delete_server(self, entry: ServerListEntry) -> None:
                         entry.server_type, account_id))
                 future = self._wallet_weakref.update_network_servers([], [], delete_server_keys, {})
                 future.result()
-
```

Comment:
This is a hack. If this change needs to be done, then do it right.

Reply:
This should be in the review, but github..

Reply:
I am assuming you want this deletion to occur in `update_network_servers`, in the callback function for updating / deleting cached values. That's what I have now done but it required synchronization to prevent a race condition with the GUI's `ServersTab.update_servers()`.